### PR TITLE
polling with pagination

### PR DESCRIPTION
**Pull request by sanjeet296**

### CODE REVIEW

File: poll.php [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+</head>
+  <body>
+    <h1> create poll </h1>
+        <hr>
+  <form  method = "post" action="poll.php">
+    <div>
+        <label>Question:</label>
+        <textarea rows="4" cols="50" name="question"></textarea>
+    </div>
+  <h2>Options</h2>
```
Comment (niteshiitj) [inline poll.php]: Format the code properly.

File: poll.php [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  <div>
+  <div>
+  <label>Option3:</label>
+  <textarea rows="4" cols="30" name="option3"></textarea>
+  </div>
+  <div>
+  <label>Option4:</label>
+  <textarea rows="4" cols="30" name="option4"></textarea>
+  </div>
+  <a href="poll.php"> Add Option </a>
```
Comment (nitesh581) [inline poll.php]: Remove it for now, will add it later

Comment (niteshiitj) [inline poll.php]: Remove Add Option feature for now, will add later

File: poll.php [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  $question=$_POST['question'];
+  $option1=$_POST['option1'];
+  $option2=$_POST['option2'];
+  $option3=$_POST['option3'];
+  $option4=$_POST['option4'];
+  $options=array( $option1, $option2, $option3, $option4);
+  $json = json_encode($options); 
+  $date=date("Y-m-d h:i:s");
+  $sql="INSERT INTO `polls`(`Question`, `Options`,`created time`) VALUES ('$question','$json','$date')";
+
```
Comment (niteshiitj) [inline poll.php]: remove unneccsary spacing

File: showpoll.php [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+?> 
+<?Php
+ 
+ if(isset($_GET['delete_id']))
+    {
+     $sql = "SELECT * FROM polls WHERE id =" .$_GET['delete_id'];
+     $result = mysqli_query($db, $sql);
+     $row = mysqli_fetch_array($result);
+    // }
+    // if(isset($_POST['btn_update'])){
```
Comment (niteshiitj) [inline showpoll.php]: remove this, unneccsary code which is of no use.

File: showpoll.php [hunk 5]
```diff
@@ -0,0 +1,317 @@
...
+
+<!DOCTYPE html>
+<html>
+<head>
+  <title>admin dashboard</title>
+  <link rel="stylesheet" type="text/css" href="style.css">
+  
+</head>
+<body>
+<a href="poll.php">poll again</a>
```
Comment (niteshiitj) [inline showpoll.php]: Rename this to 'Add New Poll'

File: showpoll.php [hunk 6]
```diff
@@ -0,0 +1,317 @@
...
+      <td>created time</td>
+      <td>lastedit time</td>
+      
+      </tr>
+      <?php
+      while($data=mysqli_fetch_assoc($run))
+
+      { 
+        $option= $data['Options'];
+        $option_s=json_decode($option);
```
Comment (niteshiitj) [inline showpoll.php]: use proper name of variable other than `option_s`

File: showpoll.php [hunk 7]
```diff
@@ -0,0 +1,317 @@
...
+        
+         
+        if( $isEditMode != $data['id']){
+         
+      ?>
+         <form method="post">
+         <tr>
+         <td><?php echo $data['id']; ?><br/><br/></td>
+         <td><?php echo $data['Question']; ?><br/><br/></td>
+         <td><?php echo $option_s[0]; ?><br/><br/></td>
```
Comment (niteshiitj) [inline showpoll.php]: use loop for showing options

File: showpoll.php [hunk 8]
```diff
@@ -0,0 +1,317 @@
...
+           }
+           else {
+             
+             
+               ?>
+            <form method="post">
+            <tr>
+            <td><?php echo $_GET['edit_id'] ?></td>
+            <td><input type="text" name="question" value="<?php echo $data['Question']; ?>" ><br/><br/></td>
+            <td><input type="text" name="option1" value="<?php echo $option_s[0]; ?>"><br/><br/></td>
```
Comment (niteshiitj) [inline showpoll.php]: Here too, always use loops instead of using index values

File: poll.php [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+  $option4=$_POST['option4'];
+  $options=array( $option1, $option2, $option3, $option4);
+  $json = json_encode($options); 
+  $date=date("Y-m-d h:i:s");
+  $sql="INSERT INTO `polls`(`Question`, `Options`,`created time`) VALUES ('$question','$json','$date')";
+  
+  
+
+  mysqli_query($db,$sql);
+  header("Location:showpoll.php?msg");
```
Comment (niteshiitj) [inline poll.php]: assign some value to msg like true/false and then use it, right now its doesn't mean anything.

File: showpoll.php [hunk 10]
```diff
@@ -119,199 +99,143 @@
...
+  $option= $data['Options'];
+  $value=json_decode($option);
+    if( $isEditMode != $data['id']){
+?>
+      <form method="post">
+      <tr>
+      <td><?php echo $data['id']; ?><br/><br/></td>
+      <td><?php echo $data['Question']; ?><br/><br/></td>
+      <?php
+      for($i=0;$i<=3;$i++){?>
```
Comment (niteshiitj) [inline showpoll.php]: don't use static length for array iteration. use count function for get array count.
Also use foreach loop.

File: showpoll.php [hunk 11]
```diff
@@ -119,199 +99,143 @@
...
-          <td><?php echo $data['last edit']; ?><br/><br/></td>
-          </tr>
-          </form>
-          <?php
-           }
+      <form method="post">
+      <tr>
+      <td><?php echo $_GET['edit_id'] ?></td>
+      <td><input type="text" name="question" value="<?php echo $data['Question']; ?>" ><br/><br/></td>
+      <td><input type="text" name="option1" value="<?php echo $value[0]; ?>"><br/><br/></td>
```
Comment (niteshiitj) [inline showpoll.php]: Maintain this section also with foreach loop.

File: poll.php [hunk 12]
```diff
@@ -1,40 +1,43 @@
...
+                <label>Option3:</label>
+                <textarea rows="4" cols="30" name="option3"></textarea>
+            </div>
+            <div>
+                <label>Option4:</label>
+                <textarea rows="4" cols="30" name="option4"></textarea>
+                <div>
+                    <button type="submit" class="btn" name="save">save</button>
+                </div>
+                <a href="admin.php">back </a>
```
Comment (niteshiitj) [inline poll.php]: div is not closing properly.

File: showpoll.php [hunk 13]
```diff
@@ -55,187 +54,206 @@
...
+        <br/>
+    </td>
+    
        
-        
-}
-      
-?>
-      
+        <?php foreach($value as $x => $val) {?>
```
Comment (niteshiitj) [inline showpoll.php]: format the code and remove all <br>

**Pull request merged**